### PR TITLE
Enhancement Updated PasswordValidator to fallback to config options (ss4)

### PR DESCRIPTION
**Pull request by flamerohr**
This pull request goes to the `4` branch and still retains instance variables to not break semver.

For more details see: https://github.com/silverstripe/silverstripe-framework/issues/7724

### CODE REVIEW

File: src/Security/PasswordValidator.php [hunk 1]
```diff
@@ -22,33 +24,73 @@ class PasswordValidator
...
+    /**
+     * @deprecated 5.0
      * Minimum password length
      *
      * @param int $minLength
      * @return $this
      */
     public function minLength($minLength)
     {
+        Deprecation::notice('5.0', 'Use ->config()->set(\'min_length\', $value) instead.');
```
Comment (tractorcow) [inline src/Security/PasswordValidator.php]: I don't think we should deprecate the instance setters; Remember that config modifications have a cost, and it's entirely possible that you could create multiple instances of PasswordValidator.

Comment (robbieaverill) [reply src/Security/PasswordValidator.php]: I'm in favour of deprecating _this one_ though, in favour of `setMinLength()` replacement

Comment (flamerohr) [reply src/Security/PasswordValidator.php]: yeah true, I'm also in favour of deprecating **this** for `setMinLength()` and other more semantic method names.

**Comment by tractorcow** [comment]
Rebasing to see if it fixes behat tests.

**Comment by tractorcow** [comment]
Versioned needs a patch https://github.com/silverstripe/silverstripe-versioned/pull/102

**Comment by tractorcow** [comment]
Behat is broken upstream, not the fault of this PR.

**Pull request merged**